RELEASE/scripts/autoscend/auto_buff.ash

```diff
@@ -142,7 +142,7 @@ boolean buffMaintain(effect buff, int mp_min, int casts, int turns, boolean spec
 	case $effect[Ashen]:					useItem = $item[pile of ashes];						break;
 	case $effect[Ashen Burps]:					useItem = $item[ash soda];						break;
 	case $effect[Astral Shell]:
-		if(acquireTotem())
+		if(auto_have_skill($skill[Astral Shell]) && acquireTotem())
```

Comment:
adding auto_have_skill here and in all other instances in buff maintain is unnecessary and redundant.
buff maintain already checks if we have the skill later on regardless of which skill it is.

if your goal is to prevent totem acquisition when we do not have ANY totem skills. then this is redundant with the changes you already made to acquireTotem()

Reply:
```suggestion
		if(acquireTotem())
```

Reply:
> adding auto_have_skill here and in all other instances in buff maintain is unnecessary and redundant. buff maintain already checks if we have the skill later on regardless of which skill it is.
> 
> if your goal is to prevent totem acquisition when we do not have ANY totem skills. then this is redundant with the changes you already made to acquireTotem()

buffMaintain(skill) checks checks if we have the skill  only after sending to acquireTotem
goal is to prevent getting totem when it isn't going to be useful, by extension the check if we do not have ANY totem skills is because  autoscend.ash sends to acquireTotem for no specific skill. putting it in acquireTotem isn't enough to remove the checks from buffMaintain(skill) because it will still want a totem for skills it doesn't have and then acquireTotem will get one if it finds any (other) totem skill

Reply:
alright. fair enough